pkg/manager/member/failure_recovery.go

```diff
@@ -0,0 +1,117 @@
+package member
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	corelisterv1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// AnnAutoFailureRecovery in TidbCluster indicates whether auto failure recovery is enabled for the components
+	AnnAutoFailureRecovery = "kubernetes.io/auto-failure-recovery"
```

Comment:
One reason (why I considered giving it a kubernetes prefix) is that the thing we are trying to handle is not entirely related to TiDB IMO and its is mostly how kubernetes deals with stateful pods. So, if this was any other stateful application like Mysql, Aerospike or Kafka we will do similar thing as well. I felt this can be a common flag name to do a common set of goals for any stateful application. I will rethink this again.

Reply:
Hi @hanlins Would you suggest having `AutoFailureRecovery` as a property in TidbCluster spec which will indicate whether auto failure recovery is enabled for the component pods, instead of having it as an annotation?

Reply:
I have no preference.

Reply:
I have renamed the annotation to "app.kubernetes.io/auto-failure-recovery"